frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/AddDataModal/Panels/CSVPanel.tsx

```diff
@@ -0,0 +1,366 @@
+              </Text>
+            </Alert>
+          </Stack>
+        ))}
+
+      {!uploadsEnabled &&
+        (canManageUploads ? (
+          <Stack gap="lg" align="center" justify="center" pt="3rem">
+            <Box component={IconCSV} c="brand" h={66} />
+            <Box component="header" ta="center">
+              <Title order={2} size="h4" mb="xs">{t`Upload CSV files`}</Title>
+              <Text maw="22.5rem" c="text-medium">
+                {c("{0} refers to the string 'your database'")
+                  .jt`To work with CSVs, enable file uploads in ${(
+                  <Tooltip
+                    inline
+                    maw="12.5rem"
+                    multiline
+                    label={t`PostgreSQL, MySQL, Redshift, and ClickHouse databases are supported for file storage.`}
+                    key="database-tooltip"
+                  >
+                    <Text td="underline">{t`your database`}</Text>
+                  </Tooltip>
+                )}.`}
+              </Text>
+              <Button
+                variant="filled"
+                w="12.5rem"
+                component={Link}
+                to={Urls.uploadsSettings()}
+              >
+                {t`Enable uploads`}
+              </Button>
+            </Box>
+          </Stack>
+        ) : (
+          <Stack
+            gap="lg"
+            align="center"
+            justify="center"
+            pt="3rem"
+            maw="22.5rem"
+          >
+            <Box component={IconCSV} c="brand" h={66} />
+            <Box component="header" ta="center">
+              <Title order={2} size="h4" mb="xs">{t`Upload CSV files`}</Title>
+              <Text c="text-medium">
+                {t`Work with CSVs, just like with any other data source.`}
+              </Text>
+            </Box>
+            <Alert icon={<Icon name="info_filled" />}>
+              <Text fz="md" lh="lg">
+                {c("${0} is admin's email address")
+                  .jt`To enable CSV file upload, please contact your administrator at ${(<b key="admin-email">{adminEmail}</b>)}.`}
+              </Text>
+            </Alert>
+          </Stack>
+        ))}
+
+      {isCollectionPickerOpen && (
+        <CollectionPickerModal
+          title={t`Select a collection`}
+          value={{ id: uploadCollectionId, model: "collection" }}
+          onChange={handleCollectionChange}
+          onClose={() => setIsCollectionPickerOpen(false)}
+          options={{
+            showPersonalCollections: true,
+            showRootCollection: true,
+            showSearch: true,
+            confirmButtonText: t`Select this collection`,
+          }}
+          models={["collection"]}
+          recentFilter={(items) =>
+            items.filter((item) => {
+              return item.model !== "table" && item.can_write;
+            })
+          }
+        />
+      )}
```

Comment:
Honestly, I don't think this API is awful. You didn't have to change anything about the collection picker to use it effectively here.

A couple weaknesses I see:
- not ideal that you have to filter out tables from recent items 🥴 
- maybe non-obvious that the collection picker can also pick dashboards (since they're collections for the purpose of questions)

cc @dpsutton